ddtrace/settings/config.py

```diff
@@ -19,10 +21,17 @@ def __init__(self):
         # use a dict as underlying storing mechanism
         self._config = {}
         self._http = HttpConfig()
+        # Master switch for turning on and off trace search by default
+        self.analytics = asbool(environ.get('DD_ANALYTICS', False))
 
     def __getattr__(self, name):
         if name not in self._config:
             self._config[name] = IntegrationConfig(self)
```

Comment:
What about:

```python
self._config[name] = IntegrationConfig(self, name)
```

Then we can handle the `get_env(self.name, 'analytics')` inside the `IntegrationConfig.__init__` ?